### PR TITLE
REF, TYP: factor out _mark_right from _add_legend_handle

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

This PR factors out adding ` (right)` to labels which are meant to go on the right y-axis. This does not change anything user-facing, and is simply a precursor to fixing #40044  and #39522 .

The change is that on `master`, only `leg.get_texts()` shows `b (right)` while `ax.right_ax.get_legend_handles_labels()` shows `b`. The former is used for constructing the user-facing legend. The reason I think it needs doing is that I believe that to address #40044  and #39522, we need to use `.get_legend_handles_labels()` for both handles and labels, while currently `.get_legend_handles_labels` is used for handles and `leg.get_texts()` is used for labels.

Putting a breakpoint after

https://github.com/pandas-dev/pandas/blob/ab687aec38668cdb0139ca02d6e37e8c8386ae7d/pandas/plotting/_matplotlib/core.py#L591

results in , on this branch:
```python
(Pdb) ax.right_ax.get_legend_handles_labels()
([<matplotlib.lines.Line2D object at 0x7fc94adf1dc0>], ['b (right)'])
```

and on master:
```python
(Pdb) ax.right_ax.get_legend_handles_labels()
([<matplotlib.lines.Line2D object at 0x7ff2892abbe0>], ['b'])
```

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
codecov shows the lines from `_mark_right_label` as not being covered - however, running `pytest pandas/tests/plotting/frame/test_frame_subplots.py -k test_subplots_dup_columns` hits them

File: pandas/plotting/_matplotlib/core.py [hunk 1]
```diff
@@ -579,11 +579,18 @@ def legend_title(self) -> Optional[str]:
             stringified = map(pprint_thing, self.data.columns.names)
             return ",".join(stringified)
 
-    def _add_legend_handle(self, handle, label, index=None):
+    def _mark_right_label(
+        self, label: Optional[str], index: Optional[int]
+    ) -> Optional[str]:
```
Comment (jreback) [inline pandas/plotting/_matplotlib/core.py:584]: can you add a doc-string

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by fangchenli** [comment]
> @fangchenli I think the coverage check here is failing because the slow plotting tests' coverage isn't currently uploaded to codecov, any suggestions for how to deal with that? Should `posix.yml` be edited so there's a step at the end which uploads to codecov, like there is in `database.yml`?

Yes, enable coverage for the build that actually run those tests and upload the report. Codecov'll automatically merge multiple reports. xref #39758

**Comment by MarcoGorelli** [comment]
codecov check passes now @jreback

File: pandas/plotting/_matplotlib/core.py [hunk 2]
```diff
@@ -579,13 +579,24 @@ def legend_title(self) -> Optional[str]:
...
+        """
+        Append ``(right)`` to the label of a line if it's plotted on the right axis.
+
+        Note that ``(right)`` is only appended when ``subplots=False``.
+        """
+        if not self.subplots and self.mark_right and self.on_right(index):
+            label += " (right)"
+        return label
+
+    def _add_legend_handle(self, handle: Artist, label: str) -> None:
```
Comment (charlesdong1991) [inline pandas/plotting/_matplotlib/core.py:592]: nitpick on the naming since we are refactoring here, maybe rephrase to `_add_legend_handle_label` or `_append_legend_handles_labels`?

**Review by charlesdong1991 — APPROVED** [review]
a minor comment on naming, could also keep as is

File: pandas/plotting/_matplotlib/core.py (hunk 2 above)
Comment (MarcoGorelli) [reply pandas/plotting/_matplotlib/core.py]: good suggestion, thanks!

**Comment by charlesdong1991** [comment]
thanks @MarcoGorelli I think it's fine to merge!
